### ISSUE TITLE
Attempting to remove an SSH key leads to a 404 error

### ISSUE THREAD

**Issue by stephenmartindale** [go-gitea/gitea#19634]
### Description

Gitea 1.16.7 (golang 1.18.1); browser: Chromium v98.

1. Navigate to the settings page -> SSH/GPG Keys.
2. Click "Remove" on any key.
3. Answer "Yes" to the modal, light-boxed pop-up.
4. Observe a **404**.
   - Interestingly, the browser appears to have redirected to `https://git.~~redacted~~.dev/user/settings/undefined` which is surely an error.
5. Observe that the SSH key was *not* removed (return to the settings page -> SSH/GPG Keys)

Obviously, I'd expect that removal of the key should work and should not throw browser errors. Addition of SSH keys does appear to work.

Apart from the 404, the only error in the browser console is the below, although I think it is unlikely to be relevant:

```
jQuery.Deferred exception: Cannot read properties of undefined (reading 'EventSourceUpdateTime') TypeError: Cannot read properties of undefined (reading 'EventSourceUpdateTime')
    at https://git.~~redacted~~.dev/_/static/assets/js/index.js?v=92fab233d575556b71e7ca75d70187aa:29:18133
    at Generator.next (<anonymous>)
    at https://git.~~redacted~~.dev/_/static/assets/js/index.js?v=92fab233d575556b71e7ca75d70187aa:29:17345
    at new Promise (<anonymous>)
    at da (https://git.~~redacted~~.dev/_/static/assets/js/index.js?v=92fab233d575556b71e7ca75d70187aa:29:17161)
    at km (https://git.~~redacted~~.dev/_/static/assets/js/index.js?v=92fab233d575556b71e7ca75d70187aa:29:18049)
    at https://git.~~redacted~~.dev/_/static/assets/js/index.js?v=92fab233d575556b71e7ca75d70187aa:62:20521
    at Generator.next (<anonymous>)
    at https://git.~~redacted~~.dev/_/static/assets/js/index.js?v=92fab233d575556b71e7ca75d70187aa:29:26285
    at new Promise (<anonymous>) undefined
s.Deferred.exceptionHook @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
Di @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
(anonymous) @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
Promise.then (async)
Lt @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
Di @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
setTimeout (async)
(anonymous) @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
_e @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
fireWith @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
fire @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
_e @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
fireWith @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
ready @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
ze @ index.js?v=92fab233d575556b71e7ca75d70187aa:62
```

It also appears that some visual glitch is defacing the pop-up mentioned in step 3, above:
![gitea-remove-ssh-key-404](https://user-images.githubusercontent.com/6370145/167113245-0bcc88cb-bc87-4c83-a7df-cc7c565d0d06.png)

The visual bug and the subsequent 404 may be related but, even if they are not, both are indicative of some bugs related to removal of an SSH key in the web interface.

### Screenshots

![gitea-remove-ssh-key-404-2](https://user-images.githubusercontent.com/6370145/167114003-716c7dd6-84c0-4b65-a71e-f4e8799e2e89.png)

### Gitea Version

1.16.7

### Can you reproduce the bug on the Gitea demo site?

No

### Operating System

Windows 10

### Browser Version

Chromium 98.0.4758.102

**Comment by stephenmartindale** [comment]
Further to last, in the `.ini`, `COOKIE_USERNAME`, `COOKIE_REMEMBER_NAME`, and `COOKIE_NAME` are customized. `COOKIE_SECURE` is set to `true`. No other documented cookie-related configuration exists.

**Comment by stephenmartindale** [comment]
I think I've found it. Something in your response made me think of that cache-clobbering token in one of the static asset URIs. I filtered network requests by that token and... without spoilers...

![image](https://user-images.githubusercontent.com/6370145/167122861-f8990527-458b-4ac0-8e97-84d1ca8ca16e.png)

Look at that `last-modified` date -- it's in January! I deployed Gitea 1.16.7 on 2022-05-03 so I'd guess that that asset is from the previous version.

So, now, I'll look in to *why* the new version of the U.I. is requesting assets with `?v=92fab233d575556b71e7ca75d70187aa` (presumably from the old version) and *why* my `nginx` server is serving them.

**Comment by stephenmartindale** [comment]
I have resolved the issue. Briefly: Gitea was being served in a Docker Compose constellation with Gitea, itself, running in one container (the back-end) and nginx (the front-end) in another. The latter would serve static assets, directly, but proxy all other requests to the Gitea back-end container. Docker Volumes were in use to convey the static assets from the Gitea image, which was built from source.

Docker Volumes are populated by their owning container if they are void when the container is initialized from its image but, when the image was updated by building a new one from source, the volume was already populated and thus not re-populated, leaving old asset files therein.

One solution would be to zap the old volume during the image update procedure but this seemed fallible so I resolved the issue by removing the use of volumes for this purpose.

Instead, I now build the image for the Gitea front-end (nginx) container with a multi-stage build that merely copies the nginx image and copies the Gitea assets into it. Here is a redacted Dockerfile:

```
ARG GITEA_TAG
ARG NGINX_TAG
FROM .../.../gitea:${GITEA_TAG} as gitea

ARG NGINX_TAG
FROM .../.../nginx:${NGINX_TAG}

COPY --from=gitea /usr/share/gitea/public/ /var/www/html
```

All I then had to do was patch the Docker compose-file to use this new image and not use volumes to populate `/var/www/html`.

My conclusion: **oops**!

Also, perhaps Gitea could or even should have some sort of safe-guards against serving old static assets with a new Gitea back-end version. Serving Gitea in this way is a documented and supported use-case for the product and the static assets are clearly self-referential because I never saw any errors from nginx or 404s related to asset files when only the old assets from 1.15 were in place -- I'd have detected the problem instantly, upon update, if the new Gitea version had tried to `GET` assets that nginx couldn't serve because the URLs simply missed.

@wxiaoguang: Thank you for your time and input -- it was very helpful. In hind-sight, this was a readily soluble issue and having worked through it has enhanced my confidence in Gitea, itself.
